### DISCUSSION TITLE
Movement at Dawn when armies present

### DISCUSSION THREAD

**Discussion by FrozenEmpire** [ChilliHugger/The-Lords-Of-Midnight#199] [Code Algorithm Questions] [answered]
Hi Chris,

Topping up the queries this morning :)

In LOM GetLocInfo(), shouldn't it be possible for a Lord to escape from a previous nights battle at Dawn ?

`
#if !defined(_DDR_)
            // if in battle there is nothing more we can do
            if ( info->foe.armies /*IsInBattle() */ )
                return info; 
#endif
`

Thanks
Andrew

**Comment by FrozenEmpire** [comment]
And when compiling for DDR, this means you can escape a battle with armies present at any time, right ?

**Comment by IcemarkUK** [comment, marked as answer]
Basically it stops the processing of other location info options - like recruitment lords, soldiers - just for LoM.
Check out here..
https://github.com/ChilliHugger/The-Lords-Of-Midnight/blob/f0242bb21c4e99e8afa61f33cf421807c217fb95/main/midnight/src/tme/scenario/item_character.cpp#L515-L521

**Reply by FrozenEmpire** [reply]
Thanks for clarifying, think I got a bit knotted up on the LOM path.

**Reply by IcemarkUK** [reply]
I still need to separate more code - need less of those #if defined

They were my quick fix when I merged the code base. Slowly working them out when I get chance...
